tools/update_dependencies.sh

```diff
@@ -14,49 +14,89 @@ function format_pyproject_toml {
     done
 }
 
+function update_npm_dependencies {
+    # Check if npm dependencies are up to date
+    echo "Updating JavaScript dependencies..." | print_info
+    npm update --save
+
+    # Fix npm security issues (skip all breaking changes)
+    echo "Running security audit of JavaScript dependencies..." | print_info
+    npm audit fix || true
+}
+
+function update_pip_dependencies {
+    # Update pip dependencies
+    echo "Updating Python dependencies..." | print_info
+    # Create temporary venv to make sure dev dependencies are not included initially
+    python3 -m venv .venv.tmp
+    source .venv.tmp/bin/activate
+    # Install package locally (without the pinned extra, so the newest available versions are installed)
+    pip install -e .
+    # Parse the newly installed versions
+    PINNED_VERSIONS=$(pip freeze --exclude-editable --local | sort)
+    PINNED_VERSIONS_TOML=$(echo "${PINNED_VERSIONS}" | format_pyproject_toml)
+    # Now also install dev dependencies
+    pip install -e .[dev]
+    # Parse the newly installed versions
+    PINNED_VERSIONS_ALL=$(pip freeze --exclude-editable --local | sort)
+    # Only consider packages that were not already pinned in the normal dependencies
+    PINNED_DEV_VERSIONS=$(comm -3 <(echo "${PINNED_VERSIONS_ALL}") <(echo "${PINNED_VERSIONS}"))
+    PINNED_DEV_VERSIONS_TOML=$(echo "${PINNED_DEV_VERSIONS}" | format_pyproject_toml)
+    # Write the new versions to pyproject.toml
+    sed --in-place --regexp-extended \
+        --expression "/^pinned = \[$/,/^\]$/c\pinned = [\n${PINNED_VERSIONS_TOML}]" \
+        --expression "/^dev-pinned = \[$/,/^\]$/c\dev-pinned = [\n${PINNED_DEV_VERSIONS_TOML}]" \
+        pyproject.toml
+    # Remove the temporary venv
+    deactivate
+    rm -rf .venv.tmp
+
+    # Install updated versions in the real venv
+    if [[ -d ".venv" ]]; then
+        # shellcheck disable=SC2102
+        pip install -e .[dev-pinned,pinned]
```

Comment:
Why is this if-else needed? Our `install.sh`-script does update dependencies, when they are updated in the pyproject.toml.

Reply:
Just a small optimization to skip the npm packages from being installed again in the `install.sh` if not needed